amqp/connection.go

```diff
@@ -2,52 +2,163 @@ package amqp
 
 import (
 	"errors"
-	"github.com/eventials/goevents/messaging"
+	"sync"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
 	amqplib "github.com/streadway/amqp"
+
+	"github.com/eventials/goevents/messaging"
 )
 
+// Connection with an AMQP peer.
 type Connection struct {
+	config     ConnectionConfig
+	m          sync.Mutex
+	url        string
 	connection *amqplib.Connection
+	closed     bool
+
+	reestablishs []chan bool
+}
+
+// ConnectionConfig to be used when creating a new connection.
+type ConnectionConfig struct {
+	reconnectInterval time.Duration
+	reconnectRetries  int
 }
 
 // NewConnection returns an AMQP Connection.
+// Uses a default ConnectionConfig with 2 second of reconnect interval and 10 retries.
 func NewConnection(url string) (messaging.Connection, error) {
```

Comment:
I don't know. Most libs I know don't take `Default`. See the streadway `Dial`, `DialTLS` and `DialConfig`.